staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go

```diff
@@ -180,38 +178,31 @@ func AddInfof(ctx context.Context, format string, data ...interface{}) {
 	}
 }
 
-func (rl *respLogger) LogArgs() []interface{} {
+// Log is intended to be called once at the end of your request handler, via defer
+func (rl *respLogger) Log() {
 	latency := time.Since(rl.startTime)
 	auditID := request.GetAuditIDTruncated(rl.req)
-	if rl.hijacked {
-		return []interface{}{
-			"verb", rl.req.Method,
-			"URI", rl.req.RequestURI,
-			"latency", latency,
-			"userAgent", rl.req.UserAgent(),
-			"audit-ID", auditID,
-			"srcIP", rl.req.RemoteAddr,
-			"hijacked", true,
+	if klog.V(3).Enabled() {
+		if !rl.hijacked {
+			klog.InfoSDepth(1, "HTTP",
+				"verb", rl.req.Method,
+				"URI", rl.req.RequestURI,
+				"latency", latency,
+				"userAgent", rl.req.UserAgent(),
+				"audit-ID", auditID,
+				"srcIP", rl.req.RemoteAddr,
+				"resp", rl.status)
+		} else {
+			klog.InfoSDepth(1, "HTTP",
+				"verb", rl.req.Method,
+				"URI", rl.req.RequestURI,
+				"latency", latency,
+				"userAgent", rl.req.UserAgent(),
+				"audit-ID", auditID,
+				"srcIP", rl.req.RemoteAddr,
+				"hijacked", true)
 		}
 	}
-	args := []interface{}{
-		"verb", rl.req.Method,
-		"URI", rl.req.RequestURI,
-		"latency", latency,
-		"userAgent", rl.req.UserAgent(),
-		"audit-ID", auditID,
-		"srcIP", rl.req.RemoteAddr,
-		"resp", rl.status,
-	}
-	if len(rl.statusStack) > 0 {
```

Comment:
What about statusStack and addedInfo args?

Reply:
Thanks @serathius for pointing it out. I have modified the logging function to include these args as well.